pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -151,7 +151,17 @@ def _assert_match(result_fill_value, expected_fill_value):
     # GH#23982/25425 require the same type in addition to equality/NA-ness
     res_type = type(result_fill_value)
     ex_type = type(expected_fill_value)
-    assert res_type == ex_type
+    if res_type.__name__ == "uint64":
+        # No idea why, but these (sometimes) do not compare as equal
```

Comment:
I believe this is numpy/numpy#12525 and [related issues](https://github.com/numpy/numpy/issues/12525#issuecomment-457727726)